src/server/http/HTTPResponder.cpp

```diff
@@ -208,6 +225,35 @@ void HTTPResponder::serveFile(HTTPClient &client, config::ServerBlock &server, c
 	client.connectionState = ASSOCIATED_FD;
 }
 
+bool HTTPResponder::checkCredentials(const std::string& authFile, const std::string& credentials) {
+	FD fd;
+	char buf[LINE_MAX];
+
+	// open and read from "password database"
+	fd = ::open(authFile.c_str(), O_RDONLY);
+	if (fd == -1)
+		ERROR_THROW(OpenFail());
+	int ret = ::read(fd, &buf, LINE_MAX);
+	if (ret == -1)
+		ERROR_THROW(ReadFail());	
+	buf[ret] = '\0';
+
+	// split per user
+	std::string file(buf); // TODO parse file?
+	std::vector<std::string> users = utils::split(file, "\n");
+	
+	// Check header
+	if (credentials.find("Basic ", 0, 6) == std::string::npos)
+		return false;
```

Comment:
check != 0 instead of == npos

otherwise header `HelloworldBasic asdfasdf` would be valid

Reply:
throw here